pkg/kubelet/status/manager.go

```diff
@@ -150,6 +154,34 @@ func (m *manager) SetPodStatus(pod *api.Pod, status api.PodStatus) {
 	m.updateStatusInternal(pod, status)
 }
 
+func (m *manager) ReconcilePodStatus(pod *api.Pod) {
+	m.podStatusesLock.Lock()
+	defer m.podStatusesLock.Unlock()
+
+	// The pod could be a mirror pod, so we should translate first.
+	uid := m.podManager.TranslatePodUID(pod.UID)
+	status, found := m.podStatuses[uid]
+	if !found {
+		// NOTE: (random-liu) What if no status? Is there any chance that this would happen?
+		glog.Warningf("Container to be reconciled has been removed in status cache: %q", format.Pod(pod))
+		return
+	}
+	if isStatusEqual(&status.status, &pod.Status) {
+		// If the status from the source is the same with the cached status,
+		// reconcile is not needed. Just return.
+		return
+	}
+	if m.needsUpdate(pod.UID, status) {
```

Comment:
This method should not be called outside of sync* functions, I think you should just drop the check from here.